app/components/message_attachments/attachment_image/attachment_image.js

```diff
@@ -48,6 +48,12 @@ export default class AttachmentImage extends PureComponent {
         }
     }
 
+    componentDidUpdate(prevProps) {
+        if (this.props.imageUrl && (prevProps.imageUrl !== this.props.imageUrl)) {
+            ImageCacheManager.cache(null, this.props.imageUrl, this.setImageUrl);
+        }
+    }
```

Comment:
Please add test.  Test case could be:
- verify if `ImageCacheManager.cache` is called and not called on props change.

Functionality looks good to me.